web-components/chart/chart.tsx

```diff
@@ -118,21 +118,31 @@ export class CobChart {
       */
       const legend = this.el.querySelector('div .mark-group .role-legend');
       const legendHeight = legend.getBoundingClientRect().height;
-      const calcHeight = this.config.height - legendHeight;
+      // We get the 'offset', or distance in px of the legend from the chart from the config.
+      // If there is no offset set, the default is 18 according to the Vega-Lite docs.
+      const offset = this.config.encoding.color.legend.offset
```

Comment:
Do you need to check to make sure everything in this chain exists?

Reply:
good call! Just updated 👍

Reply:
Well done.